### PR TITLE
SEARCH-1227/SEARCH-1232: Add Solr Affinity for node and pv

### PR DESCRIPTION
**Pull request by svidrascu**
Several changes brought by this PR:
1. Creating an additional node just for solr in same worker asg.
2. Creating an EBS volume that is used by solr only, also added two additional parameters, IndexEBSVolumeSize & IndexEBSIops, with sensible defaults.
3. Allow extra port communication between cluster nodes so that solr can communicate with repo and viceversa.
4. Tagging and tainting one of the nodes keeping in mind that the node and EBS need to be in same az.
5. Adding Additional values for SOLR.
6. Populating values for helm using values files instead of commands. This also helps to debug easier.
7. Changed deployment release names and namespace to ease up debugging.
8. Removed AvailabilityZones parameter from bastion stack since its never used anywhere in that stack.

Build -> https://bamboo.alfresco.com/bamboo/browse/PLAT-ADA51-22

### CODE REVIEW

File: scripts/helmAcs.sh [hunk 1]
```diff
@@ -127,39 +127,111 @@ EOF
...
-      --namespace=$DESIREDNAMESPACE
+
+echo "externalProtocol: https
+externalHost: \"$EXTERNAL_NAME\"
+externalPort: \"443\"
+repository:
+  adminPassword: \"$ALFRESCO_PASSWORD\"
+  image:
+    repository: \"alfresco/alfresco-content-repository-aws\"
+    tag: \"0.1.3-repo-6.0.0.3\"
```
Comment (derekhulley) [inline scripts/helmAcs.sh]: Wrong image.

File: scripts/helmAcs.sh [hunk 2]
```diff
@@ -131,74 +132,210 @@ EOF
...
 
+  # We get Bastion AZ and Region to get a valid right region and query for volumes
+  INSTANCE_ID=$(curl -s http://169.254.169.254/latest/meta-data/instance-id)
+  BASTION_AZ=$(curl -s http://169.254.169.254/latest/meta-data/placement/availability-zone)
+  REGION=${BASTION_AZ%?}
+  # We use this tag below to find the proper EKS cluster name and figure out the unique volume
+  TAG_NAME="KubernetesCluster"
+  TAG_VALUE=$(aws ec2 describe-tags --filters "Name=resource-id,Values=$INSTANCE_ID" "Name=key,Values=$TAG_NAME" --region $REGION --output=text | cut -f5)
+  # EKSname is not unique if we have multiple ACS deployments in the same cluster
+  # it must be somethign unique per Alfresco deployment, not per EKS cluster.
```
Comment (gavincornwell) [inline scripts/helmAcs.sh]: Typo In comment "somethign" -> "something"

File: scripts/helmAcs.sh [hunk 3]
```diff
@@ -131,74 +132,210 @@ EOF
...
+    initialDelaySeconds: 420
+registryPullSecrets: quay-registry-secret" >> acs_install_values.yaml
+
+    helm install alfresco-incubator/alfresco-content-services --version 1.1.5 -f acs_install_values.yaml --name $ACS_RELEASE --namespace=$DESIREDNAMESPACE
+
   fi
-  
+
   if [ "$UPGRADE" = "true" ]; then
     echo Upgrading Alfresco Content Services helm chart...
```
Comment (tunaaksoy) [inline scripts/helmAcs.sh]: (This is not a change to do as part of this PR but) do we need to specify all these values again when we do an upgrade? Can't we just use "--reuse-values" and merge in any overrides? In this case I don't even see a change in the parameters supplied. It'll reduce the maintenance.

File: scripts/helmInit.sh [hunk 4]
```diff
@@ -9,6 +9,49 @@
 export PATH=$PATH:/usr/local/bin
 export HOME=/root
 export KUBECONFIG=/home/ec2-user/.kube/config
+
```
Comment (gavincornwell) [inline scripts/helmInit.sh]: I'm not suggesting we change this now but this doesn't feel like the right place for this, this script is supposed to be for initialising helm, maybe we need an additional script/command for cluster initialisation tasks?

**Pull request merged**